### PR TITLE
expose subq-ops kernels for 1D

### PR DESCRIPTION
**Pull request by farhadrgh**
## Summary

Wires the 1D CUDA kernels from `subquadratic-ops-torch-cu12` into nvSubquadratic. Before this PR, `fft_backend="subq_ops"` only accepted 2D non-causal configs; the upstream package's 1D kernels (`fft_causal_conv1d`, `causal_conv1d`, `b2b_causal_conv1d`) were not exposed.

## What lands

**Long FFT conv (`fft_causal_conv1d`)** — wired into `CKConvND` via `fft_backend="subq_ops"` for `data_dim=1, is_causal=True`. Drop-in replacement for the `torch.fft` reference; the kernel beats the direct conv at K ≥ 128 (upstream guidance). Supports chunked and non-chunked, BHL and BLH input layouts. Shared kernel only — the upstream 1D kernel does not accept batched (per-sample / FiLM) weights, so the wrapper raises `NotImplementedError` for `B>1` kernels.

**Short causal conv (`causal_conv1d`)** — exposed as `SubqOpsCausalConv1d`, an `nn.Conv1d` subclass restricted to depthwise (`groups == in_channels == out_channels`). Drop-in for Hyena's `short_conv` slot when the host is 1D autoregressive. Supports `activation="silu"` (fused) and the standard identity path.

**b2b fused kernel (`b2b_causal_conv1d`)** — exposed as a thin pass-through function in `nvsubquadratic.ops.causal_conv1d_custom`. **Not yet wired into any module.** Designed to be the building block for a future `FusedHyena1D` variant; this PR only stabilizes the import surface.

## Capability matrix after this PR

| `fft_backend` | `data_dim` | `is_causal` | Status |
|---|---|---|---|
| `subq_ops` | 2 | False | Existed |
| `subq_ops` | 1 | True | **New** |
| `subq_ops` | 1 | False | Rejected at construction (no 1D non-causal kernel upstream) |
| `subq_ops` | 3 | * | Rejected at construction |
| `torch_fft` | * | * | Unchanged |

## Notes
- `b2b_causal_conv1d` is exposed but not module-integrated. A `FusedHyena1D` module (which would bypass `Hyena.forward` and call the fused kernel directly) is a follow-up; benchmarking the fused vs. unfused path is its own piece of work.
- No new examples / configs. Adding a striped 1D causal LM config that uses `fft_backend="subq_ops"` end-to-end is out of scope here; the smoke test in `Test1DCausalForwardBackward` is enough to verify the wiring.

## Environment setup

Create the conda environment (required to run tests):

```bash
bash setup_conda_env.sh
conda activate nvsubquadratic
```

## Pre-commit

Run locally before pushing:

```bash
pre-commit install
pre-commit run --all-files
```

### CODE REVIEW

File: nvsubquadratic/modules/ckconv_nd.py [hunk 1]
```diff
@@ -313,19 +329,35 @@ def __init__(
...
-            if use_chunked_fftconv:
-                self.fftconv_fn = fftconv2d_bhl_w_reshape_chunked
-                self.fftconv_fn_bhl_input = fftconv2d_bhl_chunked
+                if use_chunked_fftconv:
+                    self.fftconv_fn = causal_fftconv1d_bhl_w_reshape_chunked
+                    self.fftconv_fn_bhl_input = causal_fftconv1d_bhl_chunked
+                else:
+                    self.fftconv_fn = causal_fftconv1d_bhl_w_reshape
+                    self.fftconv_fn_bhl_input = causal_fftconv1d_bhl
             else:
```
Comment (Dafidofff) [inline nvsubquadratic/modules/ckconv_nd.py]: This else will break things if data-dim is set >2. Maybe change to elif?

**Review by Dafidofff — CHANGES_REQUESTED** [review]
One comment which could be improved, but further LGTM!

**Review by Dafidofff — APPROVED** [review]
LGTM!

**Comment by moradza** [comment]
@farhadrgh @Dafidofff I just remembered our fft conv2d supports both causal and non-causal. It does through an internal padding in fftconv2d.
